openid-4-verifiable-credential-issuance-1_0.md

```diff
@@ -2236,6 +2215,118 @@ The following is a non-normative example of a Credential Request with Credential
 
 The value of the `credential` claim in the Credential Response MUST be a string that is an SD-JWT VC. Credentials of this format are already suitable for transfer and, therefore, they need not and MUST NOT be re-encoded.
 
+# Claims Description {#claims-description}
+
+A claims description object is an object used to describe how a certain claim in
+the Credential should be displayed to the End-User. It is used in the `claims`
+or `credentialSubject` parameters in the Credential Issuer metadata defined in
+(#format-profiles).
+
+For a claims description object, the following keys defined by this
+specification MAY be used to describe the claim or claims (other keys MAY also
+be used):
+
+  * `path`: REQUIRED. Array. Claim path query as defined in (#claims-path-query)
+    to identify the claim(s) in the Credential.
+  * `mandatory`: OPTIONAL. Boolean which, when set to `true`, indicates that the
+    Credential Issuer will always include this claim in the issued Credential.
+    If set to `false`, the claim is not included in the issued Credential if the
+    wallet did not request the inclusion of the claim, and/or if the Credential
+    Issuer chose to not include the claim. If the `mandatory` parameter is
+    omitted, the default value is `false`.
+  * `value_type`: OPTIONAL. String value determining the type of value of the
+    claim. Valid values defined by this specification are `string`, `number`,
+    and image media types such as `image/jpeg` as defined in IANA media type
+    registry for images
+    (https://www.iana.org/assignments/media-types/media-types.xhtml#image).
+    Other values MAY also be used.
+  * `display`: OPTIONAL. Array of objects, where each object contains display
+    properties of a certain claim in the Credential for a certain language.
+    Below is a non-exhaustive list of valid parameters that MAY be included:
+     * `name`: OPTIONAL. String value of a display name for the claim.
+     * `locale`: OPTIONAL. String value that identifies language of this object
```

Comment:
Open badge v3 credentials are used extensively throughout the education sector in the US. OBV3 supports an "identity" object that contains variable data. For example
```json
{
  "identityHash": "jane@example.com",
  "identityType": "email"
}
```
but the same structure can also hold a name
```json
{
  "identityHash": "Jane Foobar",
  "identityType": "name"
}
```

Something like the below would support it, but it requires conditional logic
```json
[{
  "path": ["identifier", "identityHash"],
  "display": [{
    "name": "Name",
    "path": ["identifier", "identityType"],
    "condition": "= 'name'"
  }, {
    "name": "Email",
    "path": ["identifier", "identityType"],
    "condition": "= 'emailAddress'"
  }]
}]
```
where condition must require `[< <= = != > >=] <literal>`.
It benefit being that it can be combined with the locales

Reply:
Does this mean you have polymorphic claims?

Reply:
@paulbastian yes, and it's not just me. https://1edtech.github.io/openbadges-specification/ob_v3p0.html#identityobject

Reply:
I wouldn't support polymorphic claims, it seems this is a bad design pattern to me

Reply:
I agree that it's not a good design. However, I believe that OpenBadgeCredential v3 is one of the top 2 most popular credential types in the world. So, deciding not to support this feature in OpenID4VCI needs to be done consciously and with outreach back to 1edtech

Reply:
@paulbastian continued in https://github.com/openid/OpenID4VCI/issues/422